clients/pkg/promtail/targets/gcplog/pull_target_test.go

```diff
@@ -25,38 +25,41 @@ import (
 
 func TestPullTarget_Run(t *testing.T) {
 	// Goal: Check message written to pubsub topic is received by the target.
-	tt, apiclient, pubsubClient, teardown := testPullTarget(t)
+	ctx := context.Background()
+	tt, apiclient, pubsubClient, teardown := testPullTarget(ctx, t)
 	defer teardown()
 
 	// seed pubsub
-	ctx := context.Background()
 	tp, err := pubsubClient.CreateTopic(ctx, topic)
 	require.NoError(t, err)
+	defer tp.Stop()
+
 	_, err = pubsubClient.CreateSubscription(ctx, subscription, pubsub.SubscriptionConfig{
 		Topic: tp,
 	})
 	require.NoError(t, err)
 
 	var wg sync.WaitGroup
-
 	wg.Add(1)
+
 	go func() {
 		defer wg.Done()
-		_ = tt.run()
+		tt.run() //nolint:errcheck
 	}()
 
-	publishMessage(t, tp)
-
+	publishMessage(ctx, t, tp)
 	// Wait till message is received by the run loop.
 	// NOTE(kavi): sleep is not ideal. but not other way to confirm if api.Handler received messages
-	time.Sleep(1 * time.Second)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Do we need to change the sleep here? Shaving off 0.9s is probably not worth it if it makes the tests fail

Reply:
No, it is not necessary. But since this sleep was here to build enough time for `.Received()` to return the correct results, I split the sleep into two calls. I changed it now so that the sum of the two sleeps is equal to 1 second, WDYT?

Reply:
Sounds good :+1: thanks